plot_dataframe.py

```diff
@@ -0,0 +1,189 @@
+
+
+def parse_results_file(file):
+    """
+    Reads a log file and returns a list of lines for building the 
+    DataFrame object.
+
+    Each lines contains all the relevant hyperparameters, and the 
+    relevant data for a particular epoch.
+    """
+    df_lines = []
+    for line in file.readlines():
+        if line[:len('Namespace')] == 'Namespace':
+            params_dict = parse_params_line(line)
+            data = params_dict.copy()
+        if line[0] == '{':
+            line_dict = json.loads(line)
+            data.update({
+                key:line_dict[key]
+                for key in RELEVANT_DATA if key in line_dict
+            })
+            df_lines.append(data)
+            data = params_dict.copy()
+
+    return df_lines
+
+
+def build_dataframe(data_dir, log_file_extension=".out"):
+    all_df_lines = []
+    for file_name in os.listdir(data_dir):
+        if file_name[-len(".out"):] == log_file_extension:
+            with open(data_dir + file_name, 'r') as f:
+                df_lines = parse_results_file(f)
+                all_df_lines += df_lines
+
+    df = pd.DataFrame(all_df_lines)
+    df.set_index('epoch')
+    return df
+
+
+def plot_max_acc(df):
+    """
+    Finds the maximum accuracy over results of a sweep,
+    and display the corresponding curve
+    """
+    to_plot = df.groupby([p for p in RELEVANT_PARAMS if p not in ("random_seed")])
+
+    acc_max = 0
+    name_max = ""
+    for name, group in to_plot:
+        new_acc = group['acc_or'].groupby('epoch').mean().max()
+        if new_acc > acc_max:
+            acc_max = new_acc
+            name_max = name
+        print(name_max, acc_max)
+
+    sb.lineplot(x=to_plot.get_group(name_max)['epoch'], y=to_plot.get_group(name_max)['acc_or'], ci=None)
+    plt.title(f"Best acc : {acc_max}")
+    plt.show()
+
+
+def plot_all_curves(df, param_to_plot='acc_or', title_params=None):
+    """
+    Plot all curves of a sweep of experiments for the 
+    parameter {param_to_plot}, which needs to be in
+    RELEVANT_DATA.
+    Diplays the curves by groups of ten.
+    """
+    if param_to_plot not in RELEVANT_DATA:
+        raise ValueError(f"Argument param_to_plot should be one of "
+                         f"{RELEVANT_DATA}, not {param_to_plot}")
+
+    to_plot = df.groupby([p for p in RELEVANT_PARAMS if p not in ("random_seed")])
+
+    for i, (name, group) in enumerate(to_plot):
+        if i%10 == 0:
+            if i != 0: plt.show()
+            fig = plt.figure(figsize=(10, 5), dpi=120)
+            plt.subplots_adjust(wspace=0.5,
+                                hspace=0.5)
+        print(name)
+        ax = fig.add_subplot(2, 5, i%10+1)
+        title_indices = [RELEVANT_PARAMS.index(x)-1 for x in title_params]
+        title = '\n'.join([f"{title_params[i]}: {name[title_indices[i]]}" for i in range(len(title_params))]) 
+        ax.set_title(str(title).strip('{}'))
+        sb.lineplot(x=group['epoch'], y=group['acc_or'], ci='sd')
```

Comment:
You can also plot multiple curves on one plot with `sb.relplot`. It is used like this

```python
sb.relplot(
    kind="line",     # lineplot
    hue=XXX,         # use this to plot lines of different colors depending on parameter "XXX"  (note: you can use this with lineplot as well)
    row=YYY,         # different plot on each row depending on the value of param "YYY"
    col=ZZZ,         # different plot on each column depending on the value of param "ZZZ"
    **kwargs,         # otherwise same parameters as lineplot
)
```